pkg/analysis/duplicatemarkers/testdata/src/a/a.go

```diff
@@ -0,0 +1,69 @@
+package a
+
+// It must be ignored since it is not a type
+// +kubebuilder:validation:Enum=foo;bar;baz
+var Variable string
+
+// +kubebuilder:validation:Enum=foo;bar;baz
+// +kubebuilder:validation:Enum=foo;bar;baz
+type Enum string // want "Enum has duplicated markers kubebuilder:validation:Enum"
+
+// +kubebuilder:validation:MaxLength=10
+// +kubebuilder:validation:MaxLength=11
+type MaxLength int
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+type DuplicateMarkerSpec struct { // want "DuplicateMarkerSpec has duplicated markers kubebuilder:object:root"
+	// +kubebuilder:validation:Required
+	// shpuld be ignored since it only has single marker
+	UniqueRequired string `json:"uniqueRequired"`
+
+	// +listType=map
+	// +listMapKey=primaryKey
+	// +listMapKey=secondaryKey
+	// +required
+	// should be ignored since listMapKey is allowed to have different values
+	Map Map `json:"map"`
+
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self >= 1 && self <= 3",message="must be 1 to 5"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="replicas must be immutable"
+	// should be ignored since XValidation is allowed to have different values
+	Replicas *int `json:"replicas"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Required
+	DuplicatedRequired string `json:"duplicatedRequired"` // want "DuplicatedRequired has duplicated markers kubebuilder:validation:Required"
+
+	// +kubebuilder:validation:Enum=foo;bar;baz
+	// +kubebuilder:validation:Enum=foo;bar;baz
+	DuplicatedEnum string `json:"duplicatedEnum"` // want "DuplicatedEnum has duplicated markers kubebuilder:validation:Enum"
+
+	// +kubebuilder:validation:MaxLength=11
+	// +kubebuilder:validation:MaxLength=10
+	DuplicatedMaxlength int `json:"maxlength"`
+
+	// +listType=map
+	// +listMapKey=primaryKey
+	// +listMapKey=secondaryKey
+	// +listType=map
+	// +required
+	DuplicatedListTypeMap Map `json:"duplicatedListTypeMap"` // want "DuplicatedListTypeMap has duplicated markers listType=map"
+
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self >= 1 && self <= 3",message="must be 1 to 5"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="replicas must be immutable"
+	// +kubebuilder:validation:XValidation:rule="self >= 1 && self <= 3",message="must be 1 to 5"
```

Comment:
I wonder, does the string function on markers sort the expressions when there are multiples? CC @everettraven 

If we had 

```suggestion
	// +kubebuilder:validation:XValidation:rule="self >= 1 && self <= 3",message="must be 1 to 5"
	// +kubebuilder:validation:XValidation:message="must be 1 to 5",rule="self >= 1 && self <= 3"
```
As a separate test case, would it report a duplicate?

Reply:
I don't think it sorts them, and we don't ignore "duplicates" in our `MarkerSet` type. As far as the `MarkerSet` is concerned, we only keep track of the marker "identifier" to be able to get all instances of a particular marker identity. But we will always store all instances of the markers associated with that identity:
https://github.com/kubernetes-sigs/kube-api-linter/blob/2c83ed303cdeee7df9d6da6ea46feab0517c89ed/pkg/analysis/helpers/markers/analyzer.go#L347-L354

The comment in the linked code about the latter value taking precedence is not something I'm sure really applies anymore.

Reply:
How about checking to see if all of the markers and expressions match?